app/gui/src/project-view/util/ast/node.ts

```diff
@@ -70,14 +70,27 @@ export function primaryApplicationSubject(
   const { subject, accessChain } = Ast.accessChain(ast)
   // Require at least one property access.
   if (accessChain.length === 0) return
-  // The leftmost element must be an identifier or a placeholder.
+  const isAcceptableSubject = (subject: Ast.Ast) =>
+    (subject instanceof Ast.Ident && !subject.isTypeOrConstructor()) ||
+    subject instanceof Ast.Wildcard
   if (
-    !(
-      (subject instanceof Ast.Ident && !subject.isTypeOrConstructor()) ||
-      subject instanceof Ast.Wildcard
-    )
-  )
-    return
+    subject instanceof Ast.Group &&
+    subject.expression instanceof Ast.TypeAnnotated &&
+    isAcceptableSubject(subject.expression.expression)
+  ) {
+    return {
+      subject: subject.expression.expression.id,
+      accessChain: accessChain.map((ast) => ast.id),
+    }
+  }
+  if (subject instanceof Ast.TypeAnnotated && isAcceptableSubject(subject.expression)) {
+    return {
+      subject: subject.expression.id,
+      accessChain: accessChain.map((ast) => ast.id),
+    }
+  }
+  // The leftmost element must be an identifier or a placeholder.
+  if (!isAcceptableSubject(subject)) return
```

Comment:
I think this could be simplified by using some conditional unwrapping steps, and then the `isAcceptableSubject`/`accessChain` logic would only need to be in one place.
Something like
```ts
let innerExpression = expression
if (innerExpression instanceof Ast.Group && innerExpression.expression) innerExpression = innerExpression.expression
if (innerExpression instanceof Ast.TypeAnnotated) innerExpression = innerExpression.expression
// innerExpression can be handled uniformly
```

Reply:
Ouch, I actually missed this part of code during refactoring, thanks for spotting!

Reply:
Simplified this code greatly.